backend/src/Designer/Controllers/AppDevelopmentController.cs

```diff
@@ -158,35 +163,50 @@ public IActionResult UpdateFormLayoutName([FromBody] string newName, string org,
         }
 
         /// <summary>
-        /// Saves the layout settings
+        /// Saves the layout settings for an app without layoutsets
         /// </summary>
-        /// <param name="jsonData">The data to be saved</param>
+        /// <param name="layoutSettings">The data to be saved</param>
         /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
         /// <param name="app">Application identifier which is unique within an organisation.</param>
         /// <returns>A success message if the save was successful</returns>
         [HttpPost]
         [Route("layout-settings")]
-        public IActionResult SaveLayoutSettings([FromBody] dynamic jsonData, string org, string app)
+        public async Task<IActionResult> SaveLayoutSettings([FromBody] LayoutSettings layoutSettings, string org, string app)
         {
-            if (_repository.SaveLayoutSettings(org, app, jsonData.ToString()))
+            string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
+
+            try
             {
+                await _appDevelopmentService.SaveLayoutSettings(org, app, developer, layoutSettings, null);
                 return Ok("Layout settings successfully saved.");
             }
-
-            return BadRequest("Layout settings could not be saved.");
+            catch (FileNotFoundException exception)
+            {
+                return NotFound(exception.Message);
+            }
         }
 
         /// <summary>
-        /// Gets the layout settings
+        /// Gets the layout settings for an app without layoutSets
         /// </summary>
         /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
         /// <param name="app">Application identifier which is unique within an organisation.</param>
         /// <returns>The content of the settings file</returns>
         [HttpGet]
         [Route("layout-settings")]
-        public IActionResult GetLayoutSettings(string org, string app)
+        public async Task<ActionResult<LayoutSettings>> GetLayoutSettings(string org, string app)
         {
-            return Content(_repository.GetLayoutSettings(org, app), "application/json", Encoding.UTF8);
+            string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
+
+            try
+            {
+                LayoutSettings layoutSettings = await _appDevelopmentService.GetLayoutSettings(org, app, developer, null);
+                return Ok(layoutSettings);
+            }
+            catch (FileNotFoundException exception)
```

Comment:
Since the handling of this particular exception is repeated multiple times it's better to introduce general exception filter, and to remove try/catch from controller methods.